libretro-common/gfx/scaler/pixconv.c

```diff
@@ -249,7 +249,85 @@ void conv_rgb565_argb8888(void *output_, const void *input_,
          g            = (g << 2) | (g >> 4);
          b            = (b << 3) | (b >> 2);
 
+// Transparent RGUI hack
+#ifdef HAVE_NXRGUI
```

Comment:
This is a very specific detail of NXRGUI which does not belong here, because libretro-common is used elsewhere as well.
The specialization needs to go inside the NX code and the function should be renamed to reflect the intent here, because without extra context, it is not very obvious what the aim is.